packages/dashboard-frontend/src/components/BannerAlert/NoNodeAvailable/index.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2018-2024 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+import { api } from '@eclipse-che/common';
+import { Banner } from '@patternfly/react-core';
+import React from 'react';
+import { connect, ConnectedProps } from 'react-redux';
+
+import { container } from '@/inversify.config';
+import { WebsocketClient } from '@/services/backend-client/websocketClient';
+import { ChannelListener } from '@/services/backend-client/websocketClient/messageHandler';
+import { AppState } from '@/store';
+import { selectBranding } from '@/store/Branding/selectors';
+type Props = MappedProps;
+
+type State = {
+  startingWorkspaces: string[];
+};
+
+class BannerAlertNoNodeAvailable extends React.PureComponent<Props, State> {
+  private readonly websocketClient: WebsocketClient;
+
+  constructor(props: Props) {
+    super(props);
+    this.websocketClient = container.get(WebsocketClient);
+    this.state = {
+      startingWorkspaces: [],
+    };
+  }
+
+  public async componentDidMount() {
+    const devWorkspaceListener: ChannelListener = message => {
```

Comment:
Hi @vinokurig,
Can you use selectors for events and devWorkspaces instead of adding these WS listeners?
I mean [selectAllEvents](https://github.com/eclipse-che/che-dashboard/blob/main/packages/dashboard-frontend/src/store/Events/selectors.ts#L19) and [selectAllDevWorkspaces](https://github.com/eclipse-che/che-dashboard/blob/main/packages/dashboard-frontend/src/store/Workspaces/devWorkspaces/selectors.ts#L27)

Reply:
Hello @akurinnoy, thank you for the tip. `selectAllEvents` works for catching the `FaildedScheduling` event but there are no events related to devworkspace status, I mean we can not catch workspace started event from the array. My proposal is to keep the devworkspace event handler but to rework the namespace event handler to the `selectAllEvents` from store. WDYT?

Reply:
done, @akurinnoy please take a look